### PR TITLE
BAAAS-271 - Add oidc auth to jitrunner

### PR DESCRIPTION
**Pull request by r00ta**
https://issues.redhat.com/browse/BAAAS-271

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by r00ta** [comment]
Jenkins retest this please

File: jitexecutor/jitexecutor-runner/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,6 +1,8 @@
 quarkus.http.cors=true
 quarkus.package.type=uber-jar
 
+quarkus.oidc.enabled=false
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-runner/src/main/resources/application.properties:4]: I understand at runtime auth is NOT enabled by default 👍

File: jitexecutor/jitexecutor-runner/src/test/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,18 @@
+quarkus.http.cors=true
+
+quarkus.oidc.enabled=false
```
Comment (tarilabs) [inline jitexecutor/jitexecutor-runner/src/test/resources/application.properties:3]: test profile 👍

**Review by tarilabs — COMMENTED** [review]
Thank you @r00ta !

Do you want to write about the build command

```
mvn clean -DskipTests -Dquarkus.oidc.enabled=true -Dquarkus.http.auth.permission.authenticated.paths=/* -Dquarkus.http.auth.permission.authenticated.policy=authenticated package
```

do you want to mark it as an update to the `README.md` ?

File: jitexecutor/jitexecutor-runner/src/main/resources/application.properties (hunk 1 above)
Comment (r00ta) [reply jitexecutor/jitexecutor-runner/src/main/resources/application.properties:4]: Correct

**Comment by r00ta** [comment]
> Thank you @r00ta !
> 
> Do you want to write about the build command
> 
> ```
> mvn clean -DskipTests -Dquarkus.oidc.enabled=true -Dquarkus.http.auth.permission.authenticated.paths=/* -Dquarkus.http.auth.permission.authenticated.policy=authenticated package
> ```
> 
> do you want to mark it as an update to the `README.md` ?

Hi @tarilabs , I've updated the README accordingly :+1:

File: jitexecutor/README.md [hunk 3]
```diff
@@ -164,3 +164,18 @@ The native mode is supported by this application. You can create the native Quar
 ```bash
 mvn clean package -DskipTests -Pnative
 ```
+
+## Security
```
Comment (tarilabs) [inline jitexecutor/README.md:168]: thank you!

**Review by tarilabs — APPROVED** [review]

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
Jenkins retest this please

**Comment by r00ta** [comment]
Jenkins retest this

**Review by jiripetrlik — APPROVED** [review]

**Comment by r00ta** [comment]
@tarilabs merge please : )

**Comment by tarilabs** [comment]
@r00ta for some reason is brutally failing on Jenkins:

```
[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: JIT Executor ........................ SUCCESS [  0.052 s]

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: JIT Executor DMN .................... SUCCESS [ 37.876 s]

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: JIT Executor Runner ................. FAILURE [  0.682 s]

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests ................... SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service ... SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Common SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Quarkus SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Spring Boot SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service . SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Common SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Quarkus SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Spring Boot SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Data Index ..... SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Data Index :: Common SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Data Index :: Quarkus SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Data Index :: Spring Boot SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Task Assigning Service SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Task Assigning Service :: Processes SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Kogito Apps :: Integration Tests :: Task Assigning Service :: Quarkus SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Runtime Tools Quarkus Extension - Parent ........... SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Runtime Tools Quarkus Extension - Runtime .......... SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] Runtime Tools Quarkus Extension - Deployment ....... SKIPPED

[2021-08-31T08:40:29.066Z] [INFO] ------------------------------------------------------------------------

[2021-08-31T08:40:29.066Z] [INFO] BUILD FAILURE

[2021-08-31T08:40:29.066Z] [INFO] ------------------------------------------------------------------------

[2021-08-31T08:40:29.066Z] [INFO] Total time:  01:53 h

[2021-08-31T08:40:29.066Z] [INFO] Finished at: 2021-08-31T04:40:27-04:00

[2021-08-31T08:40:29.066Z] [INFO] ------------------------------------------------------------------------

[2021-08-31T08:40:29.066Z] ---------------------------------------------------

[2021-08-31T08:40:29.066Z] constituent[0]: file:/opt/tools/apache-maven-3.6.3/conf/logging/

[2021-08-31T08:40:29.066Z] constituent[1]: file:/opt/tools/apache-maven-3.6.3/lib/ext/takari-filemanager-0.8.3.jar

[2021-08-31T08:40:29.066Z] constituent[2]: file:/opt/tools/apache-maven-3.6.3/lib/ext/takari-local-repository-0.11.3.jar

[2021-08-31T08:40:29.066Z] constituent[3]: file:/opt/tools/apache-maven-3.6.3/lib/ext/takari-smart-builder-0.6.1.jar

[2021-08-31T08:40:29.066Z] constituent[4]: file:/opt/tools/apache-maven-3.6.3/lib/maven-embedder-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[5]: file:/opt/tools/apache-maven-3.6.3/lib/maven-settings-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[6]: file:/opt/tools/apache-maven-3.6.3/lib/maven-settings-builder-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[7]: file:/opt/tools/apache-maven-3.6.3/lib/maven-plugin-api-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[8]: file:/opt/tools/apache-maven-3.6.3/lib/maven-model-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[9]: file:/opt/tools/apache-maven-3.6.3/lib/maven-model-builder-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[10]: file:/opt/tools/apache-maven-3.6.3/lib/maven-builder-support-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[11]: file:/opt/tools/apache-maven-3.6.3/lib/maven-resolver-api-1.4.1.jar

[2021-08-31T08:40:29.066Z] constituent[12]: file:/opt/tools/apache-maven-3.6.3/lib/maven-resolver-util-1.4.1.jar

[2021-08-31T08:40:29.066Z] constituent[13]: file:/opt/tools/apache-maven-3.6.3/lib/maven-shared-utils-3.2.1.jar

[2021-08-31T08:40:29.066Z] constituent[14]: file:/opt/tools/apache-maven-3.6.3/lib/commons-io-2.5.jar

[2021-08-31T08:40:29.066Z] constituent[15]: file:/opt/tools/apache-maven-3.6.3/lib/guice-4.2.1-no_aop.jar

[2021-08-31T08:40:29.066Z] constituent[16]: file:/opt/tools/apache-maven-3.6.3/lib/guava-25.1-android.jar

[2021-08-31T08:40:29.066Z] constituent[17]: file:/opt/tools/apache-maven-3.6.3/lib/javax.inject-1.jar

[2021-08-31T08:40:29.066Z] constituent[18]: file:/opt/tools/apache-maven-3.6.3/lib/jsr250-api-1.0.jar

[2021-08-31T08:40:29.066Z] constituent[19]: file:/opt/tools/apache-maven-3.6.3/lib/plexus-utils-3.2.1.jar

[2021-08-31T08:40:29.066Z] constituent[20]: file:/opt/tools/apache-maven-3.6.3/lib/plexus-sec-dispatcher-1.4.jar

[2021-08-31T08:40:29.066Z] constituent[21]: file:/opt/tools/apache-maven-3.6.3/lib/plexus-cipher-1.7.jar

[2021-08-31T08:40:29.066Z] constituent[22]: file:/opt/tools/apache-maven-3.6.3/lib/slf4j-api-1.7.29.jar

[2021-08-31T08:40:29.066Z] constituent[23]: file:/opt/tools/apache-maven-3.6.3/lib/commons-lang3-3.8.1.jar

[2021-08-31T08:40:29.066Z] constituent[24]: file:/opt/tools/apache-maven-3.6.3/lib/maven-core-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[25]: file:/opt/tools/apache-maven-3.6.3/lib/maven-repository-metadata-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[26]: file:/opt/tools/apache-maven-3.6.3/lib/maven-artifact-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[27]: file:/opt/tools/apache-maven-3.6.3/lib/maven-resolver-provider-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[28]: file:/opt/tools/apache-maven-3.6.3/lib/maven-resolver-impl-1.4.1.jar

[2021-08-31T08:40:29.066Z] constituent[29]: file:/opt/tools/apache-maven-3.6.3/lib/maven-resolver-spi-1.4.1.jar

[2021-08-31T08:40:29.066Z] constituent[30]: file:/opt/tools/apache-maven-3.6.3/lib/org.eclipse.sisu.inject-0.3.4.jar

[2021-08-31T08:40:29.066Z] constituent[31]: file:/opt/tools/apache-maven-3.6.3/lib/plexus-component-annotations-2.1.0.jar

[2021-08-31T08:40:29.066Z] constituent[32]: file:/opt/tools/apache-maven-3.6.3/lib/maven-compat-3.6.3.jar

[2021-08-31T08:40:29.066Z] constituent[33]: file:/opt/tools/apache-maven-3.6.3/lib/plexus-interpolation-1.25.jar

[2021-08-31T08:40:29.066Z] constituent[34]: file:/opt/tools/apache-maven-3.6.3/lib/wagon-provider-api-3.3.4.jar

[2021-08-31T08:40:29.066Z] constituent[35]: file:/opt/tools/apache-maven-3.6.3/lib/org.eclipse.sisu.plexus-0.3.4.jar

[2021-08-31T08:40:29.066Z] constituent[36]: file:/opt/tools/apache-maven-3.6.3/lib/cdi-api-1.0.jar

[2021-08-31T08:40:29.066Z] constituent[37]: file:/opt/tools/apache-maven-3.6.3/lib/commons-cli-1.4.jar

[2021-08-31T08:40:29.066Z] constituent[38]: file:/opt/tools/apache-maven-3.6.3/lib/wagon-http-3.3.4-shaded.jar

[2021-08-31T08:40:29.066Z] constituent[39]: file:/opt/tools/apache-maven-3.6.3/lib/jsoup-1.12.1.jar

[2021-08-31T08:40:29.066Z] constituent[40]: file:/opt/tools/apache-maven-3.6.3/lib/jcl-over-slf4j-1.7.29.jar

[2021-08-31T08:40:29.066Z] constituent[41]: file:/opt/tools/apache-maven-3.6.3/lib/wagon-file-3.3.4.jar

[2021-08-31T08:40:29.066Z] constituent[42]: file:/opt/tools/apache-maven-3.6.3/lib/maven-resolver-connector-basic-1.4.1.jar

[2021-08-31T08:40:29.067Z] constituent[43]: file:/opt/tools/apache-maven-3.6.3/lib/maven-resolver-transport-wagon-1.4.1.jar

[2021-08-31T08:40:29.067Z] constituent[44]: file:/opt/tools/apache-maven-3.6.3/lib/maven-slf4j-provider-3.6.3.jar

[2021-08-31T08:40:29.067Z] constituent[45]: file:/opt/tools/apache-maven-3.6.3/lib/jansi-1.17.1.jar

[2021-08-31T08:40:29.067Z] ---------------------------------------------------

[2021-08-31T08:40:29.067Z] Exception in thread "main" java.lang.AssertionError

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.Assert.error(Assert.java:155)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.Assert.check(Assert.java:46)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.comp.Modules.enter(Modules.java:247)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.readSourceFile(JavaCompiler.java:837)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$ImplicitCompleter.complete(JavacProcessingEnvironment.java:1530)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Symbol.complete(Symbol.java:642)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Symbol$ClassSymbol.complete(Symbol.java:1326)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Type$ClassType.complete(Type.java:1140)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Type$ClassType.getTypeArguments(Type.java:1066)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Printer.visitClassType(Printer.java:237)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Printer.visitClassType(Printer.java:52)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Type$ClassType.accept(Type.java:993)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.code.Printer.visit(Printer.java:136)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.AbstractDiagnosticFormatter.formatArgument(AbstractDiagnosticFormatter.java:199)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.AbstractDiagnosticFormatter.formatArguments(AbstractDiagnosticFormatter.java:167)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.BasicDiagnosticFormatter.formatMessage(BasicDiagnosticFormatter.java:111)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.BasicDiagnosticFormatter.formatMessage(BasicDiagnosticFormatter.java:67)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.AbstractDiagnosticFormatter.formatArgument(AbstractDiagnosticFormatter.java:185)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.AbstractDiagnosticFormatter.formatArguments(AbstractDiagnosticFormatter.java:167)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.BasicDiagnosticFormatter.formatMessage(BasicDiagnosticFormatter.java:111)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.BasicDiagnosticFormatter.formatMessage(BasicDiagnosticFormatter.java:67)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.util.JCDiagnostic.getMessage(JCDiagnostic.java:788)

[2021-08-31T08:40:29.067Z] 	at jdk.compiler/com.sun.tools.javac.api.ClientCodeWrapper$DiagnosticSourceUnwrapper.getMessage(ClientCodeWrapper.java:799)

[2021-08-31T08:40:29.067Z] 	at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:131)

[2021-08-31T08:40:29.067Z] 	at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:174)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1134)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.plugin.compiler.TestCompilerMojo.execute(TestCompilerMojo.java:180)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)

[2021-08-31T08:40:29.067Z] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)

[2021-08-31T08:40:29.067Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

[2021-08-31T08:40:29.067Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

[2021-08-31T08:40:29.067Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

[2021-08-31T08:40:29.067Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)

[2021-08-31T08:40:29.067Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)

[2021-08-31T08:40:29.067Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)

[2021-08-31T08:40:29.067Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)

[2021-08-31T08:40:29.067Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)

script returned exit code 1
```

any idea?

**Comment by r00ta** [comment]
Jenkins run native

**Comment by domhanak** [comment]
Closing as stale, please reopen if relevant. Thanks!

**Pull request closed without merging**